src/frontend/src/tests/lib/components/guard/AdressGuard.spec.ts

```diff
@@ -0,0 +1,229 @@
+import * as btcEnv from '$env/networks.btc.env';
+import * as api from '$lib/api/backend.api';
+import { CanisterInternalError } from '$lib/canisters/errors';
+import AddressGuard from '$lib/components/guard/AddressGuard.svelte';
+import * as addressServices from '$lib/services/address.services';
+import * as authServices from '$lib/services/auth.services';
+import * as loaderServices from '$lib/services/loader.services';
+import { btcAddressMainnetStore, ethAddressStore } from '$lib/stores/address.store';
+import { authStore } from '$lib/stores/auth.store';
+import { emit } from '$lib/utils/events.utils';
+import { Ed25519KeyIdentity } from '@dfinity/identity';
+import { render } from '@testing-library/svelte';
+import { expect, type MockInstance } from 'vitest';
+
+describe('AdressGuard', () => {
+	let apiMock: MockInstance;
+
+	beforeEach(() => {
+		vi.restoreAllMocks();
+
+		vi.clearAllMocks();
+		vi.resetAllMocks();
+
+		apiMock = vi.spyOn(api, 'allowSigning');
+
+		const identity = Ed25519KeyIdentity.generate();
+		authStore.setForTesting(identity);
+
+		Object.defineProperty(window, 'location', {
+			writable: true,
+			value: {
+				href: 'https://oisy.com',
+				reload: vi.fn()
+			}
+		});
+
+		vi.spyOn(window.history, 'replaceState').mockImplementation(() => {});
+	});
+
+	describe('Signer allowance', () => {
+		it('should load signer allowance on event', async () => {
+			apiMock.mockResolvedValueOnce(undefined);
+
+			const spy = vi.spyOn(loaderServices, 'initSignerAllowance');
+
+			render(AddressGuard);
+
+			emit({ message: 'oisyValidateAddresses' });
+
+			expect(spy).toHaveBeenCalledOnce();
+		});
+
+		it('should sign out and ultimately reload the window if signer allowance fails', async () => {
+			apiMock.mockImplementation(async () => {
+				throw new CanisterInternalError('Test');
+			});
+
+			const spySignOut = vi.spyOn(authServices, 'errorSignOut');
+
+			const spy = vi.spyOn(window.location, 'reload');
+
+			render(AddressGuard);
+
+			emit({ message: 'oisyValidateAddresses' });
+
+			await vi.waitFor(() => {
+				expect(spySignOut).toHaveBeenCalledOnce();
+				expect(spy).toHaveBeenCalledOnce();
+			});
+		});
+	});
+
+	describe('Validate addresses', () => {
+		const mockEthAddress = '0x1d638414860ed08dd31fae848e527264f20512fa75d7d63cea9bbb372f020000';
+		const mockBtcAddress = '17VZNX1SN5NtKa8UQFxwQbFeFc3iqRYhem';
+
+		beforeEach(() => {
+			vi.spyOn(btcEnv, 'NETWORK_BITCOIN_ENABLED', 'get').mockReturnValue(true);
```

Comment:
Since this is not identified as a variable, when we are going to remove this flag from the project, will this be "flagged" as issue, or as unnecessary? if not, may you please add a TODO, just to keep track?

Reply:
Sure, added a TODO. Thx.